src/main/java/games/strategy/util/memento/PropertyBagMemento.java

```diff
@@ -0,0 +1,84 @@
+package games.strategy.util.memento;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Implementation of {@link Memento} that represents the originator state as a collection of name-value pairs.
+ *
+ * <p>
+ * Instances of this class are immutable.
+ * </p>
+ */
+public final class PropertyBagMemento implements Memento {
```

Comment:
Bit of a critical question, why have a Memento interface and only a PropertyBagMemento class? It seems the interface is nice in the test case, I wonder if we could use a mock instead of a fake and then combine the `Memento` interface and the single implementation. I would suggest as a name something like `GameDataMemento`, or `GameDataProxyMemento`.

What confused me for a while was I thought there would be a `PropertyBag.java` class somewhere, and was not sure whether that was a `GameData` object of some sort that we had to proxy, or if it was part of the proxy framework.  Hence I recommend consolidating the interface and single implementation together, and perhaps a name change to better capture that this is framework code.

Of note, one problem I see creeping up is an implicit definition of `originator` and `memento originator`. I do not think that term is necessarily clearly defined in one spot. To avoid this problem, I would look for public classes and methods that have in their comments `originator`. If those methods or classes were package protected, it would be pretty clearly more part of a framework, rather than some sort of 1:1 proxy or 1:1 memento that is mirroring game objects that exist somewhere else.